presto-parser/src/main/java/com/facebook/presto/sql/tree/RewritingTraversalVisitor.java

```diff
@@ -0,0 +1,491 @@
+    {
+        Expression expression = (Expression) process(node.getExpression(), context);
+        return new Cast(node.getLocation(), expression, node.getType(), node.isSafe(), node.isTypeOnly());
+    }
+
+    @Override
+    protected Node visitArithmeticBinary(ArithmeticBinaryExpression node, C context)
+    {
+        Expression expression1 = (Expression) process(node.getLeft(), context);
+        Expression expression2 = (Expression) process(node.getRight(), context);
+
+        return new ArithmeticBinaryExpression(node.getLocation(), node.getType(), expression1, expression2);
+    }
+
+    @Override
+    protected Node visitBetweenPredicate(BetweenPredicate node, C context)
+    {
+        Expression value = (Expression) process(node.getValue(), context);
+        Expression min = (Expression) process(node.getMin(), context);
+        Expression max = (Expression) process(node.getMax(), context);
+
+        return new BetweenPredicate(node.getLocation(), value, min, max);
+    }
+
+    @Override
+    protected Node visitCoalesceExpression(CoalesceExpression node, C context)
+    {
+        List<Expression> operands = processExpressionList(node.getOperands(), context);
+
+        return new CoalesceExpression(node.getLocation(), operands);
+    }
+
+    private List<Expression> processExpressionList(List<Expression> expressions, C context)
+    {
+        return ImmutableList.copyOf(Lists.transform(expressions, (value -> (Expression) process(value, context))));
+    }
+
+    @Override
+    protected Node visitArrayConstructor(ArrayConstructor node, C context)
+    {
+        List<Expression> values = processExpressionList(node.getValues(), context);
+        return new ArrayConstructor(node.getLocation(), values);
+    }
+
+    @Override
+    protected Node visitSubscriptExpression(SubscriptExpression node, C context)
+    {
+        SubscriptExpression base = (SubscriptExpression) process(node.getBase(), context);
+        Expression index = (Expression) process(node.getIndex(), context);
+
+        return new SubscriptExpression(node.getLocation(), base, index);
+    }
+
+    @Override
+    protected Node visitComparisonExpression(ComparisonExpression node, C context)
+    {
+        Expression left = (Expression) process(node.getLeft(), context);
+        Expression right = (Expression) process(node.getRight(), context);
+
+        return new ComparisonExpression(node.getLocation(), node.getType(), left, right);
+    }
+
+    @Override
+    protected Node visitQuery(Query node, C context)
+    {
+        Optional<With> with = Optional.empty();
+        if (node.getWith().isPresent()) {
+            with = Optional.of((With) process(node.getWith().get(), context));
+        }
+
+        QueryBody queryBody = (QueryBody) process(node.getQueryBody(), context);
+
+        List<SortItem> orderBy = ImmutableList.copyOf(Lists.transform(node.getOrderBy(), (value -> (SortItem) process(value, context))));
```

Comment:
We probably want to use java streams for transform.
Sth like:
```java
node.getOrderBy().stream().map(value -> (SortItem) process(value, context)).collect(toImmutableList())
```

It will probably require you to `com.facebook.presto.util.ImmutableCollectors` from `presto-main` to `presto-parser` due to lack of dependency.